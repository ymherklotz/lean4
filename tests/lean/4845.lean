/-!
  Generalize should not try to abstract the variable from hypotheses that are
  implementation details. -/

/-!
  In this case, generalize tries to revert the lemma being defined to generalize
  the 0 in it. -/

example : 0 = 0 â†’ True := by
  intro H; generalize _H : 0 = z at *
  trace_state
  constructor
