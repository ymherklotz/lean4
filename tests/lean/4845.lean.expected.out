z : Nat
_H : 0 = z
H : z = z
âŠ¢ True
